### ISSUE TITLE
Allow username in Web Chat URL

### ISSUE THREAD

**Issue by danger89** [kiwiirc/kiwiirc#1627]
**Is your feature request related to a problem? Please describe.**

I can't create a link with my username in the Web Chat URL.

**Describe the solution you'd like**

`https://web.libera.chat/<username>` (example: https://web.libera.chat/myname)

Allowing to set the username in the channel section, so you I can share a link with my IRC username.

**Describe alternatives you've considered**

`https://web.libera.chat/#myname`

But that isn't correct since it's not a channel (it should not contain the `#`). It should be a private message.

**Comment by ItsOnlyBinary** [comment]
the nick can be set in the url using the below format:

`https://web.libera.chat/?nick=mynick#mychannel`

**Comment by danger89** [comment]
> the nick can be set in the url using the below format:
> 
> `https://web.libera.chat/?nick=mynick#mychannel`

No no. I'm not talking about the nickname. I want to share a link with MY username to others, so that can start a chat with me on IRC, using their own nickname, but with my IRC name as "channel" (without #).

**Comment by PeGaSuS-Coder** [comment]
You simply can't.
If people want to talk to you directly, they just connect to a network and a channel where you're in and then they message you.

**Comment by danger89** [comment]
I know you simple can't set this now... That is why I create this feature proposal in the first place. It's added with a "new feature" label in GitHub.

We can update kiwiirc to support his.

**Comment by danger89** [comment]
@ItsOnlyBinary I don't believe that, are there no programmers in town? You can easily introduce: ?channel= query parameter. Come one.

You just don't want to implement something don't tell me it's impossible. I hate that response.

**Comment by ItsOnlyBinary** [comment]
kiwiirc uses #channel,#channel2 and as the first hash is required to make the browser treat it separate from the url it would not be possible to tell if #channel or #user

yes it would be possible to add ?channels= you also cannot have a hash in query params (unless encoded) which makes channels query param more confusing, if you was hosting your own kiwiirc you could make a plugin to give the behaviour you want but its not something I will be adding to core.

see here for example plugin: https://github.com/ItsOnlyBinary/my-kiwiirc-plugins/blob/master/plugin-urlparam-channels.js

**Comment by danger89** [comment]
Wauw. You wrote that plugin very fast indeed! Nice.

That code still needs to be adopted. But that's all right.

You could consider adding query parameter maybe called "query" to the core. So it's not confusing. Example: `?query=yournick`

**Comment by ItsOnlyBinary** [comment]
its not something i would do in core, easy to modify that plugin though ;)
